### PR TITLE
refactor(testing): just-in-time playwright test-runner builds

### LINKED ISSUE #824: Playwright dependency always fail testing
## Problem

Playwright is used to drive our website end-to-end tests.

## Steps to Reproduce

Dependency updates persistently fail. For example #819 .

## Expected behavior

When playwright dependency update is opened by Renovate, the tests run and pass, making the choice to merge clear.

## Additional context

Previous attempts to fix included grouping Playwright updates, because the library packages need to be in tandem with container image. However, this fix failed because the playwright container image used to test it is updated from main on a nightly basis.

## Proposed Solution

* Build the container image on a just-in-time basis as part of the test.
* Use layer caching to make this rebuild as fast as possible
* Use merge to main to trigger an update of the container image
* Remove existing nightly builds
* Have a way to clean-up PR builds

### PR DESCRIPTION
**Pull request by grayside**
Fixes #824 

### Approach

Build the playwright image all the time. If it doesn't have changes, it will take ~20s to load the image from cache. If it does have changes, it will take a minute or two to assemble.

### Bonus Problems Resolved

In addition to the version mismatch outlined in 824, I also found compiler target mismatch for python dependencies between the official python docker image and our test runner image. Rather than try to align two environments, I've arranged to install python dependencies inside the test runner.

### Out of Scope

As part of this change I noticed we have a scheduler job and Pub/Sub topic intended to run nightly e2e tests, but these don't appear to be used. We don't have a pub/sub or schedule-driven trigger for the e2e tests. We could get one by duplicating the PR-based trigger and adjusting the triggering event.

### Additional Context

This PR includes two detailed decision records. Please review and focus questions on comments to the decision record text.

### CODE REVIEW

**Comment by grayside** [comment]
This PR was originally opened in draft with an idea of using the playwright python container image. That image does not work for our needs (not node.js). As a result, I've continued iterating to a new solution in this branch.

File: ops/web-e2e.cloudbuild.yaml [hunk 1]
```diff
@@ -12,50 +12,72 @@
...
+      - "--cache=true"
+    dir: ops/e2e-runner
+
+  # Run E2E tests.
+  # Filesystem from previously executed steps is automatically available.
+  - id: 'Run E2E tests'
+    name: "${_REGION}-docker.pkg.dev/${PROJECT_ID}/e2e-testing/runner:kaniko-${_E2E_RUNNER_SHA}"
+    script: |
+      #!/usr/bin/env bash
+      python -m flask run --port ${_FLASK_PORT} &
```
Comment (grayside) [inline ops/web-e2e.cloudbuild.yaml:68]:
```suggestion
      python -m flask run --port ${FLASK_PORT} &
```

**Review by pattishin — APPROVED** [review]
🚀 lgtm!

File: ops/web-e2e.cloudbuild.yaml [hunk 2]
```diff
@@ -12,50 +12,72 @@
...
-      - "FLASK_PORT=${_FLASK_PORT}"
-
-# Specify a default value for optional variables
+# Specify a default value for optional variables.
 substitutions:
-  _EMBLEM_SESSION_BUCKET: nonexistent-bucket
-  _REGION: us-central1
   _E2E_RUNNER_SHA: latest
+  _EMBLEM_SESSION_BUCKET: nonexistent-bucket # no-op placeholder value
+  _EMBLEM_URL: "http://localhost:8080"
```
[the remark this answers is not in the document]
Comment (grayside) [reply ops/web-e2e.cloudbuild.yaml:76]: This substitution was previously hard-coded in the trigger, by moving the default to the cloud build.yaml it makes local testing easier.

File: docs/decisions/2023-03-31-playwright-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Chosen option [Option 1] Build playwright container as part of end-to-end test
+pipeline.
+
+This option minimizes complexity and maintenance overhead. Option 2 had promise,
+but does not include node.js runtime so we would have to rewrite our test suite.
+
+### Expected Consequences <!-- optional -->
+
+This will extend the length of time taken for the end-to-end tests, but it will
```
[the remark this answers is not in the document]
Comment (grayside) [reply docs/decisions/2023-03-31-playwright-dependencies.md:57]: Based on my trials (and you can see some data in the decision record), less than 30 seconds to load from cache, 2-3 minutes to build from a new base image.

I just realized something I didn't try: use the playwright-python image and install nodejs. Maybe that's faster to build, but that can be a future improvement.

**Pull request merged**